theseus/geometry/manifold.py

```diff
@@ -130,3 +130,10 @@ def local(
 # Alias for Manifold.retract()
 def retract(variable: Manifold, delta: torch.Tensor) -> Manifold:
     return variable.retract(delta)
+
+
+class Gaussian:
```

Comment:
Yes I think that would be a good way to add this feature. 

Precision is preferable over covariance because in GBP the messages are Gaussians over the variables and so are also represented with the Manifold class. The messages can represent relative constraints (without a global anchor) and so can have infinite uncertainty along a direction, i.e. they have rank deficient covariance matrices so we need to use the precision matrix to represent the message. Adding a positive definite check could still make sense for representing the beliefs, but for the messages I would need go disable this check.